src/renderer/components/ft-video-player/ft-video-player.js

```diff
@@ -42,6 +42,10 @@ const VHS_BANDWIDTH_VARIANCE = videojs.Vhs.BANDWIDTH_VARIANCE
 
 export default defineComponent({
   name: 'FtVideoPlayer',
+  beforeRouteLeave: function (_to, _from, next) {
+    window.removeEventListener('beforeunload', this.stopPowerSaveBlocker)
```

Comment:
I am not sure if this can/should be moved into `beforeDestroy`
But this is how it's done currently in `Watch` view

Reply:
This has to be moved to `beforeDestroy`, the route hooks only get triggered in components that are directly inside the router view. So for us those are the components in the `src/renderer/views`. Additionally the player gets destroyed and a new instance created when you change formats (maybe we can change that in the future, but for the moment that makes it easier for us), so you want this removed as soon as it's destroyed, so that you get rid of the event handler for all instances of the player.

Reply:
Updated